lldb/source/Target/SwiftLanguageRuntimeNames.cpp

```diff
@@ -393,13 +393,15 @@ SwiftLanguageRuntime::DemangleSymbolAsString(StringRef symbol, DemangleMode mode
     options.DisplayModuleNames = true;
     options.ShowPrivateDiscriminators = false;
     options.DisplayExtensionContexts = false;
+    options.DisplayLocalNameContexts = false;
```

Comment:
Why was this necessary?

Reply:
Previously, when a type alias was created inside a closure, the Swift demangler included the closure signature in the demangled type:

```
ByteBuffer #1 in closure #6 (Swift.UnsafeBufferPointer<Swift.UInt8>) -> () in main.main() -> ()
```

In my case, this was causing an assertion when fetching the Typedef CompilerType, because both signature didn't match . This was fixed with #1331 and https://github.com/apple/swift/pull/32336.

Now, by setting `DemangleOptions::DisplayLocalNameContexts` to `false`, the demangler only returns the actual type (`ByteBuffer`), avoiding the assertion.